Packs/FireEyeHX/Integrations/FireEyeHXv2/FireEyeHXv2.py

```diff
@@ -0,0 +1,2828 @@
+        try:
+            if args.get('stopSearch') == 'stop':
+                message = 'Failed to stop search'
+                client.search_stop_request(searchesIds[0])
+                message = "The search was stopped successfully"
+            # no need to stop a search before deleting it.
+            if args.get('stopSearch') == 'stopAndDelete':
+                message = 'Failed to delete search'
+                client.delete_search_request(searchesIds[0])
+                message = "The search was deleted successfully"
+        except Exception as e:
+            LOG('{}\n{}'.format(message, e))
+        commandsResults[0].readable_output += f"\n\n{message}"
+
+    return commandsResults if commandsResults else [CommandResults(readable_output="No Results")]
+
+
+def search_delete_command(client: Client, args: Dict[str, Any]) -> CommandResults:
+
+    searchIds = str(args.get('searchId')).split(',')
+
+    message = 'Results'
+    for searchId in searchIds:
+
+        try:
+            client.delete_search_request(searchId)
+            message += f'\nSearch Id {searchId}: Deleted successfully'
+        except Exception as e:
+            if '404' in str(e):
+                message += f'\nSearch Id {searchId}: Not Found'
+            else:
+                message += f'\nSearch Id {searchId}: Failed to delete search'
+
+    return CommandResults(readable_output=message)
+
+
+"""
+FETCH INCIDENT
+"""
+
+
+def fetch_incidents(client: Client, args: Dict[str, Any]) -> List:
+
+    last_run = demisto.getLastRun()
+    alerts = []  # type: List[Dict[str, str]]
+    fetch_limit = int(args.get('max_fetch') or '50')
+
+    args["sort"] = "reported_at+ascending"
+    args["limit"] = fetch_limit
+
+    # Checks if this is the first call to a function or not
+    if last_run and last_run.get('reported_at'):
+
+        # Design the filterQuery argument with last reported_at, and convert it to urlEncoding
+        query = query_fetch(reported_at=organize_reported_at(last_run.get('reported_at')))
+        args["filterQuery"] = urllib.parse.quote_plus(query)
+
+        # Get all alerts with reported_at greater than last reported_at
+        alerts = get_alerts(client, args)
+
+        # Results are sorted in ascending order - the last alert holds the greatest time *********
+        reported_at = alerts[-1].get("reported_at") if alerts else None
+
+    else:
+        # Design the filterQuery argument, and convert it to urlEncoding
+        first_fetch = args.get("first_fetch") if args.get("first_fetch") else "3 days"
+        query = query_fetch(first_fetch=first_fetch)
+        args["filterQuery"] = urllib.parse.quote_plus(query)
+
+        # Receive alerts from last 3 days - if they are more than 50 return the 50 older alerts
+        alerts = get_alerts(client, args)
+
+        # Results are sorted in ascending order - the last alert holds the greatest time
+        reported_at = alerts[-1].get("reported_at") if alerts else None
```

Comment:
This section is duplicate code, you could unite it with the code in the `if` clause above by extracting it from the `if/else` clauses.